tests/mock_network/drop.rs

```diff
@@ -25,7 +24,7 @@ fn drop_node(nodes: &mut Vec<TestNode>, index: usize) {
     let _ = poll_all(nodes);
 
     for node in nodes.iter_mut().filter(|n| close_names.contains(&n.name())) {
-        assert!(!node.inner.is_connected(name));
+        assert!(node.inner.is_peer_our_elder(&id));
```

Comment:
Shouldn't we assert that the peer is **not** our elder?

Reply:
at this moment, the dropped node is not detected by others as unresponsive, as there is not enough events accumulated. So the node still remained as elder from other peers' perspective.

Reply:
But then this check is useless because it would pass even if we didn't drop the node.

Reply:
well, I think the purpose of the test is to confirm the disconnection won't cause an immediate state change?
I agree it will pass if we don't drop a node. However, I think it could expose an issue if we changed the disconnection handling logic to `act immediately` ?

Reply:
I think we might need to rethink this test. Maybe do something like:

1. Create a network with a couple of nodes
2. Drop one node
3. Generate some traffic (user events) to let the other nodes detect the dropped node being unresponsive
4. Verify all the existing nodes detected the node being offline.

So basically the same think you do in the `verify_dropped_nodes` function in the churn tests. So maybe just extract that function into `mock_network/utils.rs` and then just call it here?

Reply:
ok. will do that

Reply:
ok. done.
However, this place is just a helper function, not a test.
so put the `verify_dropped_nodes` in the testing part.